src/bucket/BucketManagerImpl.cpp

```diff
@@ -71,6 +72,11 @@ BucketManagerImpl::initialize()
 
     mLockedBucketDir = std::make_unique<std::string>(d);
     mTmpDirManager = std::make_unique<TmpDirManager>(d + "/tmp");
+
```

Comment:
curious about your choice here: why did you hook things up at the bucket list layer instead of bucket manager? I would imagine that we should not do anything bucket related?

Reply:
`BucketManager` unfortunately doesn't only manage buckets, but `TmpDirManager` also, which is used in places like `CatchupWork` and `ProcessManager`. In general, I think disabling the bucketlist shouldn't affect temp directory creation, hence I did it this way.